scripts/o3de/o3de/utils.py

```diff
@@ -483,3 +483,36 @@ def get_object_name_and_optional_version_specifier(input:str):
         return get_object_name_and_version_specifier(input)
     except (InvalidObjectNameException, InvalidVersionSpecifierException):
         return input, None
+
+
+def replace_dict_keys_with_value_key(input:dict, value_key:str, replaced_key_name:str = None):
+    """
+    Takes a dictionary of dictionaries and replaces the keys with the value of 
+    a specific value key.
+    For example, if you have a dictionary of gem_paths->gem_json_data, this function can be used
+    to convert the dictionary so the keys are gem names instead of paths (gem_name->gem_json_data)
+    :param input: A dictionary of key->value pairs where every value is a dictionary that has a value_key
+    :param value_key: The value's key to replace the current key with
+    :param replaced_key_name: (Optional) A key name under which to store the replaced key in value
```

Comment:
Isn't this just a [flatten](https://stackoverflow.com/questions/6027558/flatten-nested-dictionaries-compressing-keys) function?

Reply:
I don't think so, the goal is not to flatten into a single dict, but to preserve the nested dict structure and replace the key in the parent dict with a value in the child.  I couldn't think of a good name for this function.  It's basically for re-keying the parent dict on `gem_name` instead of `path`, but I made it a utility function since it's seemed like a generic thing.

Reply:
So it is re-key with value function?